### PR TITLE
Update platform support matrix

### LINKED ISSUE #4220: Add IBM Z for EE as Supported Platforms
File: [engine/installation/index.md](https://docs.docker.com/engine/installation/), CC @mstanleyjones 

IBM Z is mentioned only in CE/Ubuntu here: 
https://docs.docker.com/engine/installation/#server

Need to also mention for EE for Ubuntu/RHEL/SLES
screen shot 2017-08-16 at 6 17 56 am

NOTE: For the EE 17.6.1 release, IBM Z is available only as EE Basic.

### PR DESCRIPTION
**Pull request by mstanleyjones**
Fixes #4220 

cc/ @mideidocker

### CODE REVIEW

**Comment by mstanleyjones** [comment]
Review may look gross, but this just adds a column to the table, and adds a note.

File: _includes/docker_platform_matrix.md [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+| [SUSE Linux Enterprise Server]({{ install-prefix }}/docker-ee/suse.md)     |                                                               |                                                               |                                                               | [{{ green-check }}]({{ install-prefix }}/docker-ee/suse.md)    | [{{ green-check }}]({{ install-prefix }}/docker-ee/suse.md)   |
+| [Ubuntu]({{ install-prefix }}/ubuntu.md)                                   | [{{ green-check }}]({{ install-prefix }}/docker-ce/ubuntu.md) | [{{ green-check }}]({{ install-prefix }}/docker-ce/ubuntu.md) | [{{ green-check }}]({{ install-prefix }}/docker-ce/ubuntu.md) | [{{ green-check }}]({{ install-prefix }}/docker-ee/ubuntu.md)  | [{{ green-check }}]({{ install-prefix }}/docker-ee/ubuntu.md) |
+
+> **Note**: Support for IBM Z on Docker EE is limited to Docker EE Basic.
```
Comment (joaofnfernandes) [inline _includes/docker_platform_matrix.md]: I think you can have IBM Z nodes running as managers on a plain Docker swarm, but UCP supports IBM Z nodes running as workers (the same with Windows nodes too).
@friism can you confirm that this is the case?

If so, consider changing the note to something like "With Docker EE Standard and Advanced only supports IBM Z nodes running as workers."

Comment (friism) [inline _includes/docker_platform_matrix.md]: Yes, that's correct

Comment (mstanleyjones) [inline _includes/docker_platform_matrix.md]: I tried rewording. @joaofnfernandes feel free to commit into this PR to make the wording more accurate. 👼

**Comment by joaofnfernandes** [comment]
The note is going to be weird, because the restriction itself is weird. I don't have a better proposal, so feel free to merge.

**Pull request merged**